### PR TITLE
feat(rule): add new rule 'require_scenario'

### PR DESCRIPTION
**Pull request by saw-jan**
## Description
🎉 New Rule Added: `require_scenario`

Rule metadata:
```js
meta = {
    ruleId: "require_scenario",
    type: "error"
}
```

Screenshot from 2024-04-25 21-09-21

## Related Issue
- Part of https://github.com/gherlint/gherlint/issues/17

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Tests only (no source changes)
- [ ] Documentation only (no source changes)

## Checklist:
- [ ] Code changes
- [x] Unit tests added
- [ ] Documentation updated

### CODE REVIEW

File: lib/rules/require_scenario.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        if (_isEmpty(ast?.feature)) return [];
+
+        if (RequireScenario.hasScenario(ast)) return [];
+
+        return [
+            {
+                ...RequireScenario.meta,
+                type: config.type,
+                location: { line: 1, column: 1 },
+                message: RequireScenario.meta.message,
```
Comment (saw-jan) [inline lib/rules/require_scenario.js:27]: should not be required
```suggestion
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -9,6 +9,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Added
 
+-   New lint rule `require_scenario` - enforces that a feature file has atleast one scenario (https://github.com/gherlint/gherlint/pull/87)
```
Comment (phil-davis) [inline CHANGELOG.md:12]:
```suggestion
-   New lint rule `require_scenario` - enforces that a feature file has at least one scenario (https://github.com/gherlint/gherlint/pull/87)
```

**Review by phil-davis — COMMENTED** [review]
Is there also a rule for 'scenario cannot be empty'?

Somebody could write a feature file with just:
```
Feature: the null feature

Scenario: test that the feature does nothing
```

**Comment by saw-jan** [comment]
>Is there also a rule for 'scenario cannot be empty'?

yes, we should have that one too.
Added to todo: https://github.com/gherlint/gherlint/issues/17

**Review by phil-davis — APPROVED** [review]

**Pull request merged**